docs/web/contribute/environment.md

```diff
@@ -0,0 +1,90 @@
+# Setting up a Development Environment
+
+There are two primary ways to setup a development environment.
+
+- `nix`: fewer steps, isolated
+- `conda`: more steps, not isolated
+
+## Required Dependencies
+
+- [`git`](https://git-scm.com/)
+
+## Package Management
+
+!!! warning "You need at least one package manager"
+
+    At least one of `nix` or `conda` is required to contribute to ibis.
+
+!!! info "Python 3.10 is supported on a best-effort basis"
+
+    As of 2022-01-05 there is *experimental* support for Python 3.10.
+    However, there are a number of problems with dependencies and development
+    tools that ibis uses and we cannot offically support Python 3.10 until
+    those are fixed.
+
+=== "Nix (Linux, Python 3.7-3.9)"
+
+    1. [Download and install `nix`](https://nixos.org/download.html)
+    1. Install `gh`:
```

Comment:
Is `gh` strictly needed to contribute to ibis?

Reply:
No but it's used in the later workflows to reduce the amount of information needed to contribute.